<?xml version="1.0"?>
<robot name="Volksbot"
	xmlns:xi="http://www.w3.org/2001/XInclude"
	xmlns:gazebo="http://playerstage.sourceforge.net/gazebo/xmlschema/#gz"
	xmlns:model="http://playerstage.sourceforge.net/gazebo/xmlschema/#model"
	xmlns:sensor="http://playerstage.sourceforge.net/gazebo/xmlschema/#sensor"
	xmlns:body="http://playerstage.sourceforge.net/gazebo/xmlschema/#body"
	xmlns:geom="http://playerstage.sourceforge.net/gazebo/xmlschema/#geom"
	xmlns:joint="http://playerstage.sourceforge.net/gazebo/xmlschema/#joint"
	xmlns:controller="http://playerstage.sourceforge.net/gazebo/xmlschema/#controller"
	xmlns:interface="http://playerstage.sourceforge.net/gazebo/xmlschema/#interface"
	xmlns:rendering="http://playerstage.sourceforge.net/gazebo/xmlschema/#rendering"
	xmlns:renderable="http://playerstage.sourceforge.net/gazebo/xmlschema/#renderable"
	xmlns:physics="http://playerstage.sourceforge.net/gazebo/xmlschema/#physics"
	xmlns:xacro="http://ros.org/wiki/xacro"
>

<!--Variable Pi definieren-->
<property name="pi" value="3.1415926535897931"/>
<property name="mass_base" value="25"/>     <!-- alter Wert 25 -->
<property name="mass_wheel" value="0.85"/>  <!-- alter Wert 0.75 -->

<!-- ######################  base_footprint  #######################################################
 its a fictitious link(frame) that is on the ground right below base_link origin, navigation stack depends on this frame-->
	<link name="base_footprint">
		<visual>
			<geometry>
				<box size="0.001 0.001 0.001"/>				
			</geometry>
			<origin rpy="0 0 0" xyz="0 0 0"/>
		</visual>
<!--		<collision>
			<geometry>
				<box size="0.001 0.001 0.001"/>
			</geometry>
			<origin xyz="0 0 0"/>
		</collision>
		<inertial>
			<mass value="0.0001"/>
			<inertia ixx="0.0001" ixy="0.0" ixz="0.0" iyy="0.0001" iyz="0.0" izz="0.0001"/>
			<origin xyz="0 0 0.0" />
		</inertial>  -->
	</link>
	<joint name="base_footprint_joint" type="fixed">
		<parent link="base_footprint"/>
		<child link="base_link"/>
		<origin rpy="0 0 0" xyz="0 0 0"/>
	</joint>
	
<!-- #######################  Main Body  ###################################-->	
	<link name="base_link">
		<visual>
			<geometry>
				<mesh filename="package://volksbot_new/Media/VolksbotChassis.stl"/>

			</geometry>
			<origin rpy="0 0 0" xyz="0 0 0.175"/>
			<material name="blue">
				<color rgba="0.36 .36 .36 1"/>
			</material>
		</visual>
		<collision>
			<geometry>
				<mesh filename="package://volksbot_new/Media/VolksbotChassis.stl"/>
			</geometry>
			<origin xyz="0 0 0.175"/>
		</collision>
		<inertial>
			<mass value="${mass_base}"/>
			<inertia ixx="0.83333" ixy="0.0" ixz="0.0" iyy="1.0052" iyz="0.0" izz="0.33854"/>
			<origin xyz="0 0 0.075" />
		</inertial>
	</link>
	<gazebo reference="base_link">
		<material name="Gazebo/Grey"/>
<!--		<material>Gazebo/Grey</material> -->
		<turnGravityOff>false</turnGravityOff>
	</gazebo>

<!--  ##############################  Rad vorne Rechts  ##############################-->	
	<link name="right_front_wheel">
		<visual>
			<geometry>
				<mesh filename="package://volksbot_new/Media/original5.stl"/>			
			</geometry>
			<origin rpy="${pi/2} 0 0" xyz="0 0 0"/>
			<material name="grey">
				<color rgba="0.1 0.1 0.1 1"/>
			</material>
		</visual>
		<collision>
			<geometry>
				<cylinder length="0.085" radius="0.13"/>
			</geometry>
			<origin rpy="${pi/2} 0 0" xyz="0 0 0"/>
		</collision>
		<inertial>
			<mass value="${mass_wheel}"/>
			<inertia ixx="0.0041" ixy="0.0" ixz="0.0" iyy="0.0041" iyz="0.0" izz="0.00718"/>
			<origin xyz="0 0 0.0" />
		</inertial>
	</link>

	<joint name="right_front_wheel_joint" type="continuous">
		<axis xyz="0 1 0"/>
		<parent link="base_link"/>
		<child link="right_front_wheel"/>
		<origin rpy="0 0 0" xyz="0.14 -0.2075 0.13"/>
		<limit effort="100" velocity="100"/>
		<dynamics damping="0.0" friction="0.0"/>
	</joint>
	<gazebo reference="right_front_wheel">
		<mu1 value="100"/>   <!--100-->
		<mu2 value="100"/>   <!--100-->
		<kp value="1000000.0"/> <!--100000000.0-->
		<kd value="10000.0"/>    <!--10000.0 -->
		<fdir value="1 0 0"/> 
		<friction value="100.0"/>  
		<material>Gazebo/Grey</material>
		<turnGravityOff>false</turnGravityOff>
	</gazebo>

<!--  ###############################  Rad hinten Rechts  ########################################## -->		
	<link name="right_back_wheel">
		<visual>
			<geometry>
				<mesh filename="package://volksbot_new/Media/original5.stl"/>			
			</geometry>
			<origin rpy="${pi/2} 0 0" xyz="0 0 0"/>
			<material name="grey">
				<color rgba="0.1 0.1 0.1 1"/>
			</material>
		</visual>
		<collision>
			<geometry>
				<cylinder length="0.085" radius="0.13"/>
			</geometry>
			<origin rpy="${pi/2} 0 0" xyz="0 0 0"/>
		</collision>
		<inertial>
			<mass value="${mass_wheel}"/>
			<inertia ixx="0.0041" ixy="0.0" ixz="0.0" iyy="0.0041" iyz="0.0" izz="0.00718"/>
			<origin xyz="0 0 0.0" />
		</inertial>
	</link>

	<joint name="right_back_wheel_joint" type="continuous">
		<axis xyz="0 1 0"/>
		<parent link="base_link"/>
		<child link="right_back_wheel"/>
		<origin rpy="0 0 0" xyz="-0.14 -0.2075 0.13"/>
		<limit effort="100" velocity="100"/>
		<dynamics damping="0.0" friction="0.0"/>
	</joint>
	<gazebo reference="right_back_wheel">
		<mu1 value="100"/> 
		<mu2 value="100"/>
		<kp value="1000000.0"/>  
		<kd value="10000.0"/>    
		<fdir value="1 0 0"/>
		<friction value="100.0"/>   
		<material>Gazebo/Grey</material>
		<turnGravityOff>false</turnGravityOff>
	</gazebo>

<!--  ####################################  Rad vorne Links #######################################-->	
	<link name="left_front_wheel">
		<visual>
			<geometry>
				<mesh filename="package://volksbot_new/Media/original5.stl"/>			
			</geometry>
			<origin rpy="${pi/2} 0 0" xyz="0 0 0"/>
			<material name="grey">
				<color rgba="0.1 0.1 0.1 1"/>
			</material>
		</visual>
		<collision>
			<geometry>
				<cylinder length="0.085" radius="0.13"/>
			</geometry>
			<origin rpy="${pi/2} 0 0" xyz="0 0 0"/>
		</collision>
		<inertial>
			<mass value="${mass_wheel}"/>
			<inertia ixx="0.0041" ixy="0.0" ixz="0.0" iyy="0.0041" iyz="0.0" izz="0.00718"/>
			<origin xyz="0 0 0.0" />
		</inertial>
	</link>

	<joint name="left_front_wheel_joint" type="continuous">
		<axis xyz="0 1 0"/>
		<parent link="base_link"/>
		<child link="left_front_wheel"/>
		<origin rpy="0 0 0" xyz="0.14 0.2075 0.13"/>
		<limit effort="100" velocity="100"/>
		<dynamics damping="0.0" friction="0.0"/>
	</joint>	
	<gazebo reference="left_front_wheel">
		<mu1 value="100"/>  
		<mu2 value="100"/> 
		<kp value="1000000.0"/>  
		<kd value="10000.0"/>    
		<fdir value="1 0 0"/>
		<friction value="100.0"/>  
		<material>Gazebo/Grey</material>
		<turnGravityOff>false</turnGravityOff>
	</gazebo>

<!-- #############################  Rad hinten Links  ######################################-->	
	<link name="left_back_wheel">
		<visual>
			<geometry>
				<mesh filename="package://volksbot_new/Media/original5.stl"/>			
			</geometry>
			<origin rpy="${pi/2} 0 0" xyz="0 0 0"/>
			<material name="grey">
				<color rgba="0.1 0.1 0.1 1"/>
			</material>
		</visual>
		<collision>
			<geometry>
				<cylinder length="0.085" radius="0.13"/> 	
			</geometry>
			<origin rpy="${pi/2} 0 0" xyz="0 0 0"/>
		</collision>
		<inertial>
			<mass value="${mass_wheel}"/>
			<inertia ixx="0.0041" ixy="0.0" ixz="0.0" iyy="0.0041" iyz="0.0" izz="0.00718"/>
			<origin xyz="0 0 0.0" />
		</inertial>
	</link>

	<joint name="left_back_wheel_joint" type="continuous">
		<axis xyz="0 1 0"/>
		<parent link="base_link"/>
		<child link="left_back_wheel"/>
		<origin rpy="0 0 0" xyz="-0.14 0.2075 0.13"/>
		<limit effort="100" velocity="100"/>
		<dynamics damping="0.0" friction="0.0"/>
	</joint>	
	<gazebo reference="left_back_wheel">
		<mu1 value="100"/>  // alte Werte 57
		<mu2 value="100"/> // 1.5
		<kp value="1000000.0"/>  // 1000000.0
		<kd value="10000.0"/>    // 1.0
		<fdir value="1 0 0"/>  // 1 0 0
		<friction value="100.0"/>
		<material>Gazebo/Grey</material>
		<turnGravityOff>false</turnGravityOff>
	</gazebo>	


</robot>
